packages/databricks-vscode/src/workspace-fs/WorkspaceFsAccessVerifier.ts

```diff
@@ -0,0 +1,106 @@
+import {Cluster} from "@databricks/databricks-sdk";
+import {commands, Disposable, window} from "vscode";
+import {ConnectionManager} from "../configuration/ConnectionManager";
+import {CodeSynchronizer} from "../sync";
+import {workspaceConfigs} from "../vscode-objs/WorkspaceConfigs";
+
+async function switchToRepos() {
+    await workspaceConfigs.setSyncDestinationType("repo");
+    commands.executeCommand("workbench.action.reloadWindow");
+}
+
+export class WorkspaceFsAccessVerifier implements Disposable {
+    private disposables: Disposable[] = [];
+    private currentCluster: Cluster | undefined;
+
+    constructor(
+        private _connectionManager: ConnectionManager,
+        private _sync: CodeSynchronizer
+    ) {
+        this.disposables.push(
+            this._connectionManager.onDidChangeCluster(async (cluster) => {
+                if (this.currentCluster?.name === cluster?.name) {
+                    return;
+                }
+                this.currentCluster = cluster;
+                await this.verifyCluster(cluster);
+            }),
+            this._sync.onDidChangeState(async (state) => {
+                if (
+                    workspaceConfigs.syncDestinationType === "repo" &&
+                    state === "FILES_IN_REPOS_DISABLED"
+                ) {
+                    await window.showErrorMessage(
+                        "Sync failed. Files in Repos is disabled for the current workspace. Please contact your system admin to enable it for your workspace."
```

Comment:
Can we link to docs that describe how to do it?
Alternatively we might want to offer a deep link into workspace settings.